src/com/github/kornilova203/matlab/psi/MatlabResolvingScopeProcessor.kt

```diff
@@ -14,9 +14,9 @@ class MatlabResolvingScopeProcessor(private val myReference: MatlabReference) :
 
     override fun execute(decl: PsiElement, state: ResolveState): Boolean {
         if (decl !is MatlabDeclaration) return true
-        if (decl.name == myReference.element.text) {
+        if (decl.name == myReference.element.text && (decl is MatlabGlobalVariableDeclaration || declaration == null || declaration?.firstChild == myReference.element)) {
```

Comment:
Let's extract the last part to a function `isResolvedToItself`.
Note that I used `identifyingElement` function because identifier element doesn't have to be the first child 
```kotlin
    private fun isResolvedToItself(): Boolean {
        return declaration?.identifyingElement == myReference.element.identifier()
    }
```